src/Surfnet/ServiceProviderDashboard/Application/Service/EntityMergeService.php

```diff
@@ -101,8 +104,12 @@ public function mergeEntityCommand(SaveEntityCommandInterface $command, ?ManageE
         }
 
         $oidcClient = null;
+        $isNewEntity = !$manageEntity;
+        $isCopyToProduction = $manageEntity && $manageEntity->getId() === null;
+
         $secret = new NullSecret();
-        if (!$manageEntity && $protocol->getProtocol() !== Constants::TYPE_SAML) {
+        // Set the client secret when createing a new entity or when we are copying an entity.
+        if ($protocol->getProtocol() !== Constants::TYPE_SAML && ($isNewEntity || $isCopyToProduction)) {
```

Comment:
You could consider lowering the cyclomatic complexity by moving this check to a method on the entity itself. Something like `hasClientSecret` for example. Also when new entity types are introduced you need to implement this check and it would be harder to forget to update this check. But feel free to leave it as is for now.